app/models/claim/advocate_claim.rb

```diff
@@ -272,12 +272,24 @@ def instantiate_basic_fees
 
     def destroy_all_invalid_fee_types
       if case_type.present? && case_type.is_fixed_fee?
-        basic_fees.map(&:clear) unless basic_fees.empty?
+        clear_basic_fees
+        destroy_ineligbile_fixed_fees
       else
         fixed_fees.destroy_all unless fixed_fees.empty?
       end
     end
 
+    def clear_basic_fees
+      basic_fees.map(&:clear) unless basic_fees.empty?
+    end
+
+    def destroy_ineligbile_fixed_fees
+      eligbile_fees = Claims::FetchEligibleFixedFeeTypes.new(self).call
+      fixed_fees.each do |fee|
+        fixed_fees.delete(fee) unless eligbile_fees.map(&:code).include?(fee.fee_type_code)
```

Comment:
```
fee.delete unless....
```
also: are we wanting to perform a delete instead of a destroy deliberately? [delete versus destroy](https://stackoverflow.com/questions/22757450/difference-between-destroy-and-delete)

```
# ensures callbacks are handled (dates_attended?)
fee.destroy unless...
```

Reply:
It wasn't a conscious decision on my part, but I think the version without callbacks is preferable... happy to discuss...

Reply:
it looks like `association.delete(child)` delegates its strategy to the associations `:dependant` . so `fixed_fees.delete(fee)` will call :destroy on the fee and therefore destroy on its children
[rails association delete](https://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-delete)

so `fixed_fees.delete(fee)` is the same as `fixed_fee.destroy(fee)`. `fee.delete` deletes the fee but not its children, `fee.destoy` deletes its children but cached results are not removed so   calling `claim.fixed_fees.flat_map(&:dates_attended).size` fails but `claim.fixed_fees.flat_map(&:dates_attended).count` does not

so better to use `association.delete(child)` or if we want to be explicit and not delegate strategy to the relation then `association.destroy(child)`. delegation magic is probably the "rails" way to go. 

I found it interesting anyway, and have added specs for it.